homeassistant/components/generic/camera.py

```diff
@@ -125,32 +125,45 @@ def camera_image(self):
         ).result()
 
     async def async_camera_image(self):
+        """Wrap _async_camera_image with an asyncio.shield."""
+        # Shield the request because of https://github.com/encode/httpx/issues/1461
+        try:
+            self._last_url, self._last_image = await asyncio.shield(
+                self._async_camera_image()
+            )
+        except asyncio.CancelledError as err:
+            _LOGGER.warning("Timeout getting camera image from %s", self._name)
+            raise err
+        return self._last_image
+
+    async def _async_camera_image(self):
         """Return a still image response from the camera."""
         try:
             url = self._still_image_url.async_render(parse_result=False)
         except TemplateError as err:
             _LOGGER.error("Error parsing template %s: %s", self._still_image_url, err)
-            return self._last_image
+            return self._last_url, self._last_image
 
         if url == self._last_url and self._limit_refetch:
-            return self._last_image
-
+            return self._last_url, self._last_image
+        response = None
         try:
             async_client = get_async_client(self.hass, verify_ssl=self.verify_ssl)
             response = await async_client.get(
                 url, auth=self._auth, timeout=GET_IMAGE_TIMEOUT
             )
             response.raise_for_status()
-            self._last_image = response.content
+            image = response.content
         except httpx.TimeoutException:
             _LOGGER.error("Timeout getting camera image from %s", self._name)
-            return self._last_image
+            return self._last_url, self._last_image
         except (httpx.RequestError, httpx.HTTPStatusError) as err:
             _LOGGER.error("Error getting new camera image from %s: %s", self._name, err)
-            return self._last_image
-
-        self._last_url = url
-        return self._last_image
+            return self._last_url, self._last_image
+        finally:
+            if response:
+                response.close()
```

Comment:
Wait. Isn't close async here for an async httpx instance? Ie it should await response.aclose() also this is generally not needed unless it's a stream=true as mentioned here 
https://www.python-httpx.org/async/#opening-and-closing-clients

Reply:
You are right, I think it should be async close..I didn't change it when we switched back from the sync API.
Let me check into the close thing. I think when I was looking at the issues over on httpx there were other issues with not closing and freeing up resources...

Reply:
I think issue 116/147 and pr 117 over there mean we should try to close it ourselves

Reply:
If you don't have any objections, I'm going to go ahead and merge this as is. Hopefully the underlying `httpx` issue gets taken care of soon and we can roll back the shielding/closing code.
BTW, in https://github.com/home-assistant/core/issues/47624 someone noted that there may be similar issues with `rest`. It is possible that there are similar problems in other components that use `httpx`.